### ISSUE TITLE
URL().exists() would be cool

### ISSUE THREAD

**Issue by michaelaye** [aio-libs/yarl#780]
### Is your feature request related to a problem?

For debugging purposes to quickly check if building up an URL tree is correct, a `exists()` method like pathlib.Path has would be nice.

### Describe the solution you'd like

Check an URL for existence (i.e. net availability) by using ULR().exists().

I am currently using something like this, and have found that this is a a very quick way doing this:

```python
def check_url_exists(url):
    response = requests.head(url)
    if response.status_code < 400:
        return True
    else:
        return False
```

### Describe alternatives you've considered

None.

### Code of Conduct

- [X] I agree to follow the aio-libs Code of Conduct

**Comment by webknjaz** [comment]
Thanks for the suggestion! Unfortunately, yarl is not the best place for implementing this. Its only purpose is to provide URL representation and manipulate parts of it. That's it.

TL;DR this is one of the scope.

YARL does not intend to perform any I/O. Moreover, pathlib relies on synchronous I/O while yarl is mostly catering frameworks of the asynchronous world. So it's be rather `await URL.exists()` and would require the user to set up the asyncio loop properly beforehand. Besides, it would mean that it's need a mechanism for configuring HTTP requests because not all the sites would allow HEAD requests. I'd also create a circular dependency on aiohttp which isn't great either.
